packages/openid4vc/src/shared/utils.ts

```diff
@@ -49,21 +51,62 @@ async function getKeyFromDid(agentContext: AgentContext, didUrl: string) {
   return getKeyFromVerificationMethod(verificationMethod)
 }
 
-export function getVerifyJwtCallback(agentContext: AgentContext): VerifyJwtCallback {
+type VerifyJwtCallbackOptions = {
+  federation?: {
+    trustedEntityIds?: string[]
+  }
+}
+
+export function getVerifyJwtCallback(
+  agentContext: AgentContext,
+  options: VerifyJwtCallbackOptions = {}
+): VerifyJwtCallback {
   return async (jwtVerifier, jwt) => {
     const jwsService = agentContext.dependencyManager.resolve(JwsService)
+
     if (jwtVerifier.method === 'did') {
       const key = await getKeyFromDid(agentContext, jwtVerifier.didUrl)
       const jwk = getJwkFromKey(key)
 
       const res = await jwsService.verifyJws(agentContext, { jws: jwt.raw, jwkResolver: () => jwk })
       return res.isValid
-    } else if (jwtVerifier.method === 'x5c' || jwtVerifier.method === 'jwk') {
+    }
+
+    if (jwtVerifier.method === 'x5c' || jwtVerifier.method === 'jwk') {
       const res = await jwsService.verifyJws(agentContext, { jws: jwt.raw })
       return res.isValid
-    } else {
-      throw new Error(`Unsupported jwt verifier method: '${jwtVerifier.method}'`)
     }
+
+    if (jwtVerifier.method === 'openid-federation') {
```

Comment:
We need to start splitting I think between:
- verifying and making sure that the context is ok (e.g. what we do with dids, we just check that the did is the one that signed)
- verifying and make sure that the context is trusted (what we now do with x509 and federation in this callback). 

I think to make things more customizable we should allow the frist case as well, so the part of just verifing the signature against the identifiers, is different from whether the identifier is trusted. Does that make sense?

not sure how to act on this now, but it also related to an earlier comment on the more unified trust callback (whether you're using jwk / did / x5c / federation).

Reply:
I think this is a great idea but also requires a bit more thinking and I see this more as a improvement in the near future.